cmake/sample_defs/sample_mission_cfg.h

```diff
@@ -64,7 +64,7 @@
 **      MESSAGE_FORMAT_IS_CCSDS_VER_2 is optional
 */
 /* #define MESSAGE_FORMAT_IS_CCSDS_VER_2 */
-#undef MESSAGE_FORMAT_IS_CCSDS_VER_2
+#define MESSAGE_FORMAT_IS_CUSTOM
```

Comment:
Why would we need a compile time switch for custom/not custom header format?  Every mission needs one (and only one) defined message format.  It should just be "the selected message format".  

I assume "not custom" means the traditional CCSDS v1 with CFE-defined secondary headers, using the 6-byte TLM header?  Does that mean every user who wants v2/extended header needs to enable this "custom" switch?

I'd rather see something where the user just specifies one specific header format, by name.  If that happens to be one that CFE provides out of the box, then great.  Otherwise the user will have to provide an implementation to go with their selection.

Reply:
You want to break everyone's config if they don't define version 1?  We certainly can take that approach if it's preferred.  I went with minimizing impacts (it works just like default (v1), define for V2 as before, but adds an additional IS_CUSTOM).

Could also have unique typedefs for each option, and just define as a common name here.  Is that what you are asking for?

Reply:
No it doesn't break existing config because it can be done on the CMake side.  If CMake needs to do some sort of source selection based on the header format then CMake needs to know what the user wants, so it can select the files appropriately.

So it would work just like anything else - CMake has a default, which is used if the user does nothing, and if the user sets something then it uses that.

Besides that, even if it was only in the C header file, it's just a matter of putting the default value into the sample file.  (but again, setting in the value via CMake is totally backward compatible provided we give a default)

Reply:
Yes, it could certainly be done multiple ways and I was hoping for a CMake related update (per the design suggestions I provided) but I haven't seen progress on that front in line with our allotted schedule.  Given that, I made these trivial changes that also work to redefine the header.  This is certainly not meant as the perfect solution, it's just a trivial solution that works so I can still make schedule if a better solution is not complete in time.